nomad/upload_metrics.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# this script is executed on Nomad clients, after a scenario scenario has been run, to upload metrics with telegraf.
+# It also takes care of setting the RUN_ID if unset from env
+
+set -euo pipefail
+
+function check_envset() {
+  local var_name="$1"
+  # temporarily allow fail to bind
+  set +u
+  if [[ -z "${!var_name}" ]]; then
+    echo "Environment variable $var_name is not set." >&2
+    set -u
+    exit 1
+  fi
+  set -u
+}
```

Comment:
Couldn't we just first check if it's set so that we don't have to use `set +u/-u`?

```suggestion
function check_envset() {
  local var_name="$1"
  if [[ ! -v "$var_name" ]] || [[ -z "${!var_name}" ]]; then
    echo "Environment variable $var_name is not set or is an empty string." >&2
    exit 1
  fi
}
```

Reply:
@cdunster I've reverted. It would be good, but I've just found out, just because I was testing another issue on macos, that on MacOS bash 3.2 is installed, which doesn't support -v.